### PR TITLE
interop server: support secure connections

### PR DESCRIPTION
**Pull request by jtattermusch**
- import test creds normally used for interop testing
- support `--port` and `--use_tls` cmdline args.

### CODE REVIEW

File: testassets/InteropTestsWebsite/Resources.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+using System.IO;
+
+namespace InteropTestsWebsite
+{
+    public static class Resources
+    {
+        public static string CertDir = Path.Combine(GetSolutionDirectory(), "testassets/InteropTestsWebsite", "Certs");
+        public static string ServerPFXPath = Path.Combine(CertDir, "server1.pfx");
+
+        private static string GetSolutionDirectory()
```
Comment (jtattermusch) [inline testassets/InteropTestsWebsite/Resources.cs]: logic copied from examples for now, but I think a want to use a simple way.

File: testassets/InteropTestsWebsite/Program.cs [hunk 2]
```diff
@@ -29,16 +31,29 @@ public static void Main(string[] args)
...
-        public static IWebHostBuilder CreateWebHostBuilder(string[] args) =>
-            WebHost.CreateDefaultBuilder(args)
-                .ConfigureKestrel(options =>
+        public static IWebHostBuilder CreateWebHostBuilder(string[] args)
+        {
+            var builder = WebHost.CreateDefaultBuilder(args);
+
+            // Support --port and --use_tls cmdline arguments normally supported
+            // by gRPC interop servers.
+            int port = int.Parse(builder.GetSetting("port") ?? "50052");
```
Comment (jtattermusch) [inline testassets/InteropTestsWebsite/Program.cs]: not really proud of this, but it works. Is there a more idiomatic way? :-)

Comment (JunTaoLuo) [reply testassets/InteropTestsWebsite/Program.cs]: [Command line configuration](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-2.2#command-line-configuration-provider)! This is added by default by the WebHost.CreateDefaultBuilder.

File: testassets/InteropTestsWebsite/Certs/ca.pem [hunk 3]
```diff
@@ -0,0 +1,15 @@
+-----BEGIN CERTIFICATE-----
```
Comment (JunTaoLuo) [inline testassets/InteropTestsWebsite/Certs/ca.pem]: Maybe we can use these certs for the examples as well. Less duplication and infrastructure.

File: testassets/InteropTestsWebsite/Resources.cs (hunk 1 above)
Comment (JunTaoLuo) [reply testassets/InteropTestsWebsite/Resources.cs]: Yea, duplication is always undesired, not a big deal though.

Comment (JamesNK) [reply testassets/InteropTestsWebsite/Resources.cs]: I'm not a fan of this way of referencing certs. Is there a reason why we have this pattern?

I'd prefer that we include the certs are content in the project, copy them to the output directory on build and reference them there.

Should change this both for the samples and the interop test site.

File: testassets/InteropTestsWebsite/Program.cs (hunk 2 above)
Comment (JunTaoLuo) [reply testassets/InteropTestsWebsite/Program.cs]: Something like 
```c#
                .ConfigureKestrel((context, options) =>
                {
                    context.Configuration.GetValue<int>("port");
                    ...
                }
```

File: testassets/InteropTestsWebsite/Resources.cs (hunk 1 above)
Comment (JunTaoLuo) [reply testassets/InteropTestsWebsite/Resources.cs]: I recall trying this when I set this up but I didn't get it to work. Copying to the output directory means the files will be in the `bin/Debug/netcoreapp3.0` folder. However, when you invoke `dotnet run`, the root directory is set to the project directory, which means you need to include `bin/Debug/netcoreapp3.0` in your path to the certificate file. We found that resolving the full path more robust (against Debug/Release for example). 

Is there any new csproj mechanisms that I'm not aware of that can make this work?

Comment (JamesNK) [reply testassets/InteropTestsWebsite/Resources.cs]: Lets keep going with this for now Because We Know It Works, and clean it up later.

Comment (JamesNK) [reply testassets/InteropTestsWebsite/Resources.cs]: https://github.com/grpc/grpc-dotnet/issues/94

File: testassets/InteropTestsWebsite/Certs/ca.pem (hunk 3 above)
Comment (jtattermusch) [reply testassets/InteropTestsWebsite/Certs/ca.pem]: sg, let's address in a follow up PR?

File: testassets/InteropTestsWebsite/Program.cs (hunk 2 above)
Comment (jtattermusch) [reply testassets/InteropTestsWebsite/Program.cs]: Done. Thanks for help!

**Comment by jtattermusch** [comment]
I think this should be good to merge. PTAL.

File: testassets/InteropTestsWebsite/Program.cs [hunk 4]
```diff
@@ -16,9 +16,12 @@
 
 #endregion
 
+using System.Runtime.InteropServices;
+
```
Comment (JamesNK) [inline testassets/InteropTestsWebsite/Program.cs]: nit: remove extra line

**Pull request merged**